client/js/widgets/widget.js

```diff
@@ -942,6 +942,24 @@ export class Widget extends StateManaged {
         }
         if((a.property == 'parent' || a.property == 'deck') && a.value !== null && !widgets.has(a.value)) {
           problems.push(`Tried setting ${a.property} to ${a.value} which doesn't exist.`);
+        } else if (a.property == 'id' && isValidCollection(a.collection)) {
+          if (collections[a.collection].length) {
+            const oldWidget = collections[a.collection][0];
+            var newState = Object.assign({},oldWidget.state);
```

Comment:
This shallow copy of state is probably adequate, but I've implemented a deep copy of state just in case.

Reply:
I actually think this should be a shallow copy, since any children of children will be keeping their old parent. (When you say "I've implemented", you mean you did that somewhere else, right?)

Reply:
> I actually think this should be a shallow copy, since any children of children will be keeping their old parent. (When you say "I've implemented", you mean you did that somewhere else, right?)

No this is in my PR too (initializing newState by parsing oldState). `widget.state` does not contain any information about the children.

Reply:
Yeah, that should be a deep copy. Otherwise changes to array properties will jump to the other widget.

Like David says, the `state` does not contain children. It is "exactly" what you see in the JSON editor.